src/Symfony/Component/Ldap/Security/LdapUserProvider.php

```diff
@@ -144,7 +144,7 @@ public function supportsClass($class)
     /**
      * Loads a user from an LDAP entry.
      *
-     * @return LdapUser
+     * @return LdapUser|UserInterface
```

Comment:
why both? The real return type would still be one-of

Reply:
> Declaration of Symfony\Component\Security\Core\User\LdapUserProvider::loadUser($username, Symfony\Component\Ldap\Entry $entry): Symfony\Component\Security\Core\User\User must be compatible with Symfony\Component\Ldap\Security\LdapUserProvider::loadUser($username, Symfony\Component\Ldap\Entry $entry): Symfony\Component\Ldap\Security\LdapUser in /var/www/html/symfony/src/Symfony/Component/Security/Core/User/LdapUserProvider.php on line 55

Reply:
i see, missed that. But this new ldap provider we aim for `loadUser(): UserInterface` in 5.0 isnt it? shouldnt we hint as such already?

Reply:
right, updated